frontend/src/app/main/data/workspace/libraries.cljs

```diff
@@ -701,14 +714,19 @@
    (ptk/reify ::launch-component-sync
      ptk/WatchEvent
      (watch [_ state _]
-       (let [current-file-id (:current-file-id state)
+       (let [data (get state :workspace-data)
+             component (ctkl/get-component data component-id)
+             page (ctf/get-component-page data component)
+             root (ctf/get-component-root data component)
```

Comment:
Yo don't need to get the whole page and root objects. You could just take `:main-instance-id` and `:main-instance-page` attributes of the component, and send them to `dwt/update-thumbnail`.